src/QsCompiler/CSharpGeneration/SimulationCode.fs

```diff
@@ -389,7 +389,8 @@ module SimulationCode =
         let rec buildExpression (ex: TypedExpression) =
             match ex.Expression with
             // TODO: Diagnostics
-            | InvalidExpr -> failwith "Can't generate code for error expression"
+            | InvalidExpr -> failwith "Can't generate code for invalid expression."
+            | Lambda _ -> failwith "Can't generate code for un-lifted lambda."
```

Comment:
Nit: 
This code is a part of the tool that outputs the diagnostic messages, like "Can't generate code for invalid expression". E.g. this tool may be a compiler console application. 
Sometimes such tools are invoked by other tools, like IDE. E.g. the VSCode can be invoking a compiler console application.
Those other tools (VSCode) may want to intercept those diagnostic messages ("Can't generate code for invalid expression") and show them to the user in a different form, e.g. surrounded with parentheses, angular brackets, quotes or single quotes, e.g. 
`Compiler application exited with error (message: <Can't generate code for invalid expression>)`

If your diagnostic messages end with punctuation marks (in this case dot `.`) then it will be inconvenient to surround such message with some other characters. So the message above will turn to - note the dot in the end of the diagnostic message:
`Compiler application exited with error (message: <Can't generate code for invalid expression.>)`

You may want to consider removing the punctuation marks in the end of the diagnostic message string literals:
"Can't generate code for invalid expression." (last dot `.` before closing quote)

Reply:
`failwith` throws an exception, and from what I've seen of exception messages from .NET, they are by convention complete sentences that end in a period.